### PR TITLE
Plot computes extents instead of Dataset.

### PR DESCRIPTION
**Pull request by jtlan**
`_getExtent()` and `_computeExtent()` removed from `Dataset`.

### CODE REVIEW

**Comment by aicioara** [comment]
Will do in the morning

**Comment by bluong** [comment]
It looks like functionality is mostly kept except there is no caching.  Will there be any caching in this PR?

**Comment by jtlan** [comment]
Cacheing is already there: the `_attrToExtents` field caches the extent for a given attribute and is only updated with `_updateExtentsForAttr()`.

File: test/core/metadataTests.ts [hunk 1]
```diff
@@ -188,27 +188,6 @@ describe("Metadata", () => {
...
-      };
-    };
-    plot.addDataset(dataset);
-    plot.renderTo(svg);
-    assert.deepEqual(dataset._getExtent(a, id), [16, 17], "plot metadata is reflected in extent results");
-    dataset.metadata({foo: 0});
-    assert.deepEqual(dataset._getExtent(a, id), [5, 6], "plot metadata is reflected in extent results after change user metadata");
-    svg.remove();
-  });
-
```
Comment (aicioara) [inline test/core/metadataTests.ts]: Shouldn't this functionality be now tested in `plotTests.ts`? in particular the `plot._computeExtent()` method

Comment (jtlan) [inline test/core/metadataTests.ts]: `_computeExtent()` is private. There are a number of indirect tests of it by adding test data and seeing the resultant domains of the scales, which I think is better.

Comment (aicioara) [inline test/core/metadataTests.ts]: Fair enough

**Pull request merged**